include/ethash/ethash.h

```diff
@@ -49,6 +49,9 @@ extern "C" {
 #define ETHASH_FULL_DATASET_ITEM_SIZE 128
 #define ETHASH_NUM_DATASET_ACCESSES 64
 
+/// The maximum epoch number supported by this implementation.
+#define ETHASH_MAX_EPOCH_NUMBER 32639
```

Comment:
How did you arrive at this number?

Reply:
The max value where light cache size is under 4GB. Later some uint32 overflows start to happen.

Reply:
@axic
1 epoch = 30k blocks so the limit imposed by @chfast is 979.17M blocks
Difficulty is not related. Difficulty is adjusted in mining to keep block interval constant (among 12 and 15 seconds).
Actually mainnet is at block 13.5M i.e. epoch 450

Reply:
This is documented now.

Reply:
@AndreaLanfranchi @chfast Thanks for the explanation, this looks good now.